core/src/main/java/co/cask/wrangler/utils/StructuredRecordConverter.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.wrangler.utils;
+
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.format.UnexpectedFormatException;
+import co.cask.cdap.api.data.schema.Schema;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+
+import java.io.IOException;
+import java.lang.reflect.Array;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Converts an {@link StructuredRecord} to subset of the input {@link StructuredRecord} based
+ * on the {@link Schema} specified.
+ */
+public final class StructuredRecordConverter {
+
+  public static StructuredRecord transform(StructuredRecord record, Schema schema) throws IOException {
+    StructuredRecord.Builder builder = StructuredRecord.builder(schema);
+    for (Schema.Field field : record.getSchema().getFields()) {
+      String name = field.getName();
+      builder.set(name, convertField(record.get(name), field.getSchema()));
```

Comment:
If the field is missing from the provided output `schema`, setting it would result in failure.